source/common/router/router.cc

```diff
@@ -1380,6 +1405,13 @@ void Filter::onUpstreamData(Buffer::Instance& data, UpstreamRequest& upstream_re
   // streams.
   ASSERT(upstream_requests_.size() == 1);
   if (end_stream) {
+    Upstream::ClusterRequestResponseSizeStatsOptRef req_resp_stats =
+        cluster_->requestResponseSizeStats();
+    if (req_resp_stats.has_value()) {
+      req_resp_stats->get().upstream_rs_body_size_.recordValue(
+          callbacks_->streamInfo().bytesSent());
+    }
```

Comment:
We never buffer the upstream at this point, so this too would only account for the last data chunk

Reply:
Same as above, thought callbacks_->streamInfo().bytesSent() would be cumulative